src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/controllers/LibraryControllerTests.kt

```diff
@@ -1,153 +1,175 @@
+import org.springframework.boot.autoconfigure.couchbase.CouchbaseProperties.Env
 import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
 import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.context.annotation.Import
-
+import org.springframework.test.context.TestPropertySource
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
 import org.springframework.test.web.servlet.setup.MockMvcBuilders
 
 @SpringBootTest
 @AutoConfigureMockMvc
 @Import(LibraryController::class)
-class LibraryControllerTests() {
-    lateinit var libraryService: LibraryService
-    lateinit var requestBuilder: RequestBuilder
-    var endpoint = "/libraries/"
-    var id1 = "70b664a416135a6e967fadc6"
-    var id2 = "dd7f03b962f1f3416d08ee0f"
-    @BeforeEach
-    fun configureSystem() {
-        libraryService = mock(LibraryService::class.java)
-        val libraryController = LibraryController(libraryService)
-        val mockMvc = MockMvcBuilders.standaloneSetup(libraryController)
-                .build()
-        requestBuilder = RequestBuilder(mockMvc)
+@TestPropertySource(properties = ["CLIENT_ID=test_id", "CLIENT_SECRET=test_secret"])
+class LibraryControllerTests {
+  lateinit var libraryService: LibraryService
+  lateinit var requestBuilder: RequestBuilder
+  var endpoint = "/libraries/"
+  var id1 = "70b664a416135a6e967fadc6"
+  var id2 = "dd7f03b962f1f3416d08ee0f"
+
+  @BeforeEach
+  fun configureSystem() {
+    libraryService = mock(LibraryService::class.java)
+    val libraryController = LibraryController(libraryService)
+    val mockMvc = MockMvcBuilders.standaloneSetup(libraryController)
+      .build()
+    requestBuilder = RequestBuilder(mockMvc)
+  }
+
+  @Nested
+  @DisplayName("Tests for getLibraries")
+  inner class GetLibraries {
+    /** Tests returning a 200 response for returning all libraries */
+    @Test
+    fun shouldReturnAllLibraries() {
+      val library1 = Library("id1", "Backlog", ArrayList())
+      val library2 = Library("id2", "Owned Games", ArrayList())
+      library2.games.add("gameId1")
+      library2.games.add("gameId2")
+      val libraries = ArrayList<Library>()
+      libraries.add(library1)
+      libraries.add(library2)
+      given(libraryService.getAll()).willReturn(libraries)
+      requestBuilder.runGetRequest(endpoint)
+        .andExpect(status().isOk)
+        .andExpect(jsonPath("$[0].name", equalTo("Backlog")))
+        .andExpect(jsonPath("$[0].games", equalToObject(ArrayList<String>())))
+        .andExpect(jsonPath("$[0].id", equalTo("id1")))
+        .andExpect(jsonPath("$[1].name", equalTo("Owned Games")))
+        .andExpect(jsonPath("$[1].games[0]", equalTo("gameId1")))
+        .andExpect(jsonPath("$[1].games[1]", equalTo("gameId2")))
+        .andExpect(jsonPath("$[1].id", equalTo("id2")))
     }
+    /** Tests returning 204 for getting all libraries */
+    @Test
+    fun shouldReturnNoValues() {
```

Comment:
When would we expect a user to have no libraries? Should we allow this?

Reply:
Yeah that's a good point. I was trying to be exhaustive with the test cases but this would never really happen as we have the required libraries that shouldn't be able to be deleted. I'll remove the test